extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -105,15 +115,39 @@ CamelRegistryBuildItem registry(
             return new CamelRegistryBuildItem(registry);
         }
 
+        @Overridable
+        @BuildStep
+        public CamelModelJAXBContextFactoryBuildItem createJaxbContextFactory() {
+            return new CamelModelJAXBContextFactoryBuildItem(new DisabledModelJAXBContextFactory());
+        }
+
+        @Overridable
+        @BuildStep
+        public CamelXmlLoaderBuildItem createXmlLoader() {
+            return new CamelXmlLoaderBuildItem(new DisabledXmlLoader());
+        }
+
+        @BuildStep
+        void disableXmlReifiers(List<FeatureBuildItem> features) {
+            if (features.stream().map(FeatureBuildItem::getInfo).noneMatch("camel-xml"::equals)) {
+                ProcessorReifier.registerReifier(ValidateDefinition.class, DisabledValidateReifier::new);
+                ValidatorReifier.registerReifier(PredicateValidatorDefinition.class, DisabledPredicateValidatorReifier::new);
```

Comment:
I guess this set-up should be done from a recorder otherwise such changes are lost

Reply:
I'm not sure about that.  The code changes a static variable (the map of reifiers) which is later stored in the native image.  There's no need to call anything at a later point I think.

Reply:
for my understanding, how this end up being in the generated classes ? otherwise my understanding (that could be of course wrong) is that it is gogin to be ignored